### PR TITLE
VB-1748 move prisoner profile page requests to orchestration service

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
Add prisoner profile page endpoint to orchestration service

### CODE REVIEW

**Comment by dhirajshettymoj** [comment]
Draft PR request for review. Some work is pending on making the calls to endpoints asynchronously.

**Review by AledWEvans — CHANGES_REQUESTED** [review]
The code is good code a few comments, The thing is I'm noticing names of properties that are different for the same things offenderNo, prisonerId, prisonerNumber, convention dictates these should be all name the same thing in our system,.. but what we are doing is importing the names from the third party systems and therefor breaking encapsulation rules and allowing legacy inconstancies in our system.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/whereabouts/ScheduledEventDto.kt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  val eventTypeDesc: @NotBlank String? = null,
+
+  @Schema(required = true, description = "Sub type (or reason) of scheduled event (as a code)")
+  val eventSubType: @NotBlank String? = null,
+
+  @Schema(required = true, description = "Description of scheduled event sub type")
+  val eventSubTypeDesc: @NotBlank String? = null,
+
+  @Schema(required = true, description = "Date on which event occurs")
+  val eventDate: @NotNull LocalDate? = null,
```
[the remark this answers is not in the document]
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/whereabouts/ScheduledEventDto.kt]: I have updated it based on the Schema here - https://whereabouts-api-dev.service.justice.gov.uk/swagger-ui/index.html to make only bookingId a NotNull field. Currently the WhereAbouts API call is not integrated. When we integrate the WhereAbouts API call we will drop the fields not needed and will revisit this DTO then.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/PrisonerProfileDto.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.prison.api.AlertDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.prison.api.InmateDetailDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.prison.api.PrisonerBookingSummaryDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.prison.api.VisitBalancesDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.prisoner.search.PrisonerDto
+import java.time.LocalDate
+
+data class PrisonerProfileDto(
+  @Schema(required = true, description = "Prisoner Number", example = "A1234AA")
+  val offenderNo: String,
```
[the remark this answers is not in the document]
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/PrisonerProfileDto.kt]: Updated to prisonerId in almost all instances except where the Dto is for an external client response.

**Review by AledWEvans — APPROVED** [review]
LGTM

**Pull request merged**